framework/integration/test/reconciliation/reconciliation_test.go

```diff
@@ -1,21 +1,113 @@
-// +build k8srequired
-
 package reconciliation
 
 import (
+	"reflect"
 	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/giantswarm/operatorkit/framework/integration/client"
 )
 
 // Test_Finalizer_Integration_Reconciliation is a integration test for
 // the proper replay and reconciliation of delete events with finalizers.
 func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
+	configMapName := "test-cm"
+	testFinalizer := "operatorkit.giantswarm.io/test-operator"
 	testNamespace := "finalizer-integration-reconciliation-test"
+	testOtherFinalizer := "operatorkit.giantswarm.io/other-operator"
+	operatorName := "test-operator"
 
 	client.MustSetup(testNamespace)
 	defer client.MustTeardown(testNamespace)
 
-	// TODO: Implement the actual test here.
+	operatorkitFramework, err := client.NewFramework(operatorName, testNamespace)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We create a configmap, but add a finalizer of another operator. This will
+	// cause the configmap to continue existing after the framework removes it own
+	// finalizer.
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: testNamespace,
+			Finalizers: []string{
+				testOtherFinalizer,
+			},
+		},
+		Data: map[string]string{},
+	}
+	_, err = client.CreateConfigMap(cm)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We start the framework.
+	go operatorkitFramework.Boot()
```

Comment:
This is my headache right now, I can't think of an idea to get proper feedback from our framework. The informer starts it watch inside the framework, so I couldn't get a handle on the channels - is there a way to do this?

Reply:
https://github.com/giantswarm/operatorkit/pull/182/files#r178519161